app/src/views/private/components/overlay-item.vue

```diff
@@ -0,0 +1,625 @@
+			if (newActive) internalActive.value = true;
+			else cancel();
+		},
+	});
+
+	return { save, cancel, overlayActive };
+
+	function save() {
+		const editsToValidate = props.junctionField ? internalEdits.value[props.junctionField] : internalEdits.value;
+		const fieldsToValidate = props.junctionField ? relatedCollectionFields.value : fieldsWithoutCircular.value;
+		const defaultValues = getDefaultValuesFromFields(fieldsToValidate);
+		const existingValues = props.junctionField ? initialValues?.value?.[props.junctionField] : initialValues?.value;
+
+		const errors = validateItem(
+			merge({}, defaultValues.value, existingValues, editsToValidate),
+			fieldsToValidate,
+			isNew.value,
+		);
+
+		if (nestedValidationErrors.value?.length) errors.push(...nestedValidationErrors.value);
+
+		if (errors.length > 0) {
+			validationErrors.value = errors;
+			return;
+		} else {
+			validationErrors.value = [];
+		}
+
+		if (props.junctionField && Object.values(defaultValues.value).some((value) => value !== null)) {
+			internalEdits.value[props.junctionField] = internalEdits.value[props.junctionField] ?? {};
+		}
+
+		if (props.junctionField && props.relatedPrimaryKey !== '+' && relatedPrimaryKeyField.value) {
+			set(internalEdits.value, [props.junctionField, relatedPrimaryKeyField.value.field], props.relatedPrimaryKey);
+		}
+
+		if (props.primaryKey && props.primaryKey !== '+' && primaryKeyField.value) {
+			internalEdits.value[primaryKeyField.value.field] = props.primaryKey;
+		}
+
+		emit('input', internalEdits.value);
+
+		internalActive.value = false;
+		internalEdits.value = {};
+	}
+
+	function cancel() {
+		validationErrors.value = [];
+		internalActive.value = false;
+		internalEdits.value = {};
+	}
+}
+</script>
+
+<template>
+	<v-drawer
+		v-if="overlay === 'drawer'"
+		v-model="overlayActive"
+		:title="title"
+		:icon="collectionInfo?.meta?.icon ?? undefined"
+		persistent
+		@cancel="cancel"
+	>
+		<template v-if="template !== null && templateData && primaryKey !== '+'" #title>
+			<v-skeleton-loader v-if="loading || templateDataLoading" class="title-loader" type="text" />
+
+			<h1 v-else class="type-title">
+				<render-template :collection="templateCollection?.collection" :item="templateData" :template="template" />
+			</h1>
+		</template>
+
+		<template #subtitle>
+			<v-breadcrumb :items="[{ name: collectionInfo?.name, disabled: true }]" />
+		</template>
+
+		<template #actions>
+			<slot name="actions" />
+
+			<v-button v-tooltip.bottom="t('save')" icon rounded :disabled="!isSavable" @click="save">
+				<v-icon name="check" />
+			</v-button>
+		</template>
+
+		<overlay-item-content
+			v-model:internal-edits="internalEdits"
+			:collection
+			:primary-key
+			:junction-field
+			:related-collection
+			:initial-values
+			:fields
+			:disabled
+			:loading
+			:validation-errors
+			:junction-field-location
+			:related-collection-fields
+			:related-primary-key
+			:refresh
+			class="drawer-item-content"
+		/>
```

Comment:
This is repeated three times with one prop difference, could move it to a <component> and computed property